web/src/users/mod.rs

```diff
@@ -0,0 +1,73 @@
+use axum::{
+    extract::FromRequestParts,
+    http::request::Parts,
+    response::{IntoResponse, Response},
+};
+use cja::app_state::AppState as _;
+
+use thiserror::Error;
+
+pub use db::users::User;
+
+use crate::{
+    sessions::{Session, SessionError},
+    AppState,
+};
+
+pub mod login;
+pub mod signup;
+
+#[derive(Error, Debug)]
+pub enum ExtractUserError {
+    #[error("No session")]
+    NoSession,
+    #[error("No user id")]
+    NoUserId,
+    #[error("SQL error: {0}")]
+    SqlxError(#[from] sqlx::Error),
+}
+
+impl IntoResponse for ExtractUserError {
+    fn into_response(self) -> Response {
+        match self {
+            Self::NoSession | Self::NoUserId => {
+                axum::response::Redirect::temporary("/login").into_response()
+            }
+            Self::SqlxError(_) => axum::http::StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+        }
+    }
+}
+
+impl From<SessionError> for ExtractUserError {
+    fn from(_: SessionError) -> Self {
+        ExtractUserError::NoSession
+    }
+}
+
+#[async_trait::async_trait]
+impl FromRequestParts<AppState> for User {
+    type Rejection = ExtractUserError;
+
+    async fn from_request_parts(
```

Comment:
This is what allows us to ask for a `user: User` in our handlers!

We take the session and look for a `user_id` to load